backend/api/api_test.go

```diff
@@ -0,0 +1,356 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/gin-gonic/contrib/cors"
+	"github.com/gin-gonic/gin"
+	"github.com/stretchr/testify/assert"
+)
+
+var router *gin.Engine
+
+// Print the DB
+func displayTable() {
+	rows, err := db.Query("SELECT * FROM list")
+	if err != nil {
+		fmt.Println(err.Error())
+	}
+
+	defer rows.Close()
+
+	// Get all rows and add into items
+	items := make([]ListItem, 0)
+	for rows.Next() {
+		// Individual row processing
+		item := ListItem{}
+		if err := rows.Scan(&item.Id, &item.Item, &item.Done); err != nil {
+			fmt.Println(err.Error())
+		}
+		items = append(items, item)
+	}
+
+	fmt.Println("items:", items)
+}
+
+// Delete all elements
+// from DB
+func emptyTable() {
+	db.Exec("DELETE from list;")
+
+	// Reset id counter
+	db.Exec("ALTER SEQUENCE list_id_seq RESTART WITH 1;")
+}
+
+// Setup Gin Routes
+func SetupRoutes() *gin.Engine {
+	// Use Gin as router
+	router := gin.Default()
+	config := cors.DefaultConfig()
+	config.AllowAllOrigins = true
+	router.Use(cors.New(config))
+
+	// Set routes for API
+	router.GET("/items", TodoItems)
+	router.GET("/item/create/:item", CreateTodoItem)
```

Comment:
It would be better to use POST for create, PUT for update and DELETE For delete operation.

Reply:
True. I had set configured them as POST, PUT, and DELETE first but found issues with the React frontend performing fetch-api calls to the API backend. This is why I had configured all the routes as GET.